FWCore/Framework/src/SubProcess.cc

```diff
@@ -557,19 +580,28 @@ namespace edm {
   void SubProcess::doStreamBeginRunAsync(WaitingTaskHolder iHolder,
                                          unsigned int id,
                                          RunPrincipal const& principal,
-                                         IOVSyncValue const& ts) {
+                                         IOVSyncValue const& ts,
+                                         std::vector<std::shared_ptr<const EventSetupImpl>> const* iEventSetupImpls) {
     typedef OccurrenceTraits<RunPrincipal, BranchActionStreamBegin> Traits;
 
     RunPrincipal& rp = *principalCache_.runPrincipalPtr();
 
-    beginStreamTransitionAsync<Traits>(
-        std::move(iHolder), *schedule_, id, rp, ts, esp_->eventSetup(), serviceToken_, subProcesses_);
+    beginStreamTransitionAsync<Traits>(std::move(iHolder),
+                                       *schedule_,
+                                       id,
+                                       rp,
+                                       ts,
+                                       esp_->eventSetupImpl(),
```

Comment:
Why not get the specific `EventSetupImpl` from `iEventSetupImpls`?

Reply:
See comment above